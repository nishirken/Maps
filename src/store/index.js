export testInitialState from './test-initial-state';
export testServerInitialState from './test-server-initial-state';
export store from './create-store';
export testStore from './test-store';
