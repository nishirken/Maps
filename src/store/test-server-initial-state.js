import { fromJS } from 'immutable';

export default fromJS([
    {
        index: 1,
        coords: {
            lng: 100,
            lat: 200,
            x: 20,
            y: 21,
        },
        name: 'Marker 1',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
    {
        index: 2,
        coords: {
            lng: 200,
            lat: 300,
            x: 10,
            y: 11,
        },
        name: 'Marker 2',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
]);
