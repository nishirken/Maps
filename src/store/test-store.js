import configureStore from 'redux-mock-store';
import initialState from './test-initial-state';

const mockStore = configureStore();
const store = mockStore(initialState.toObject());

export default store;
