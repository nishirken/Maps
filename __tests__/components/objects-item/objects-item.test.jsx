import store from 'Store';
import { ObjectsItem } from 'Components';

describe('Objects item component', () => {
    const ObjectsItemShallow = shallow(<ObjectsItem store={store} />);
    const ObjectsItemMount = mount(<ObjectsItem store={store} />);

    it('Rendered', () => {
        expect(ObjectsItemShallow).toMatchSnapshot();
    });

    it('Has a delete button', () => {
        expect(ObjectsItemMount.find('svg').length).toBe(1);
    });

    it('Has an inner text', () => {
        expect(ObjectsItemMount.text().length).toBeGreaterThan(1);
    });
});
