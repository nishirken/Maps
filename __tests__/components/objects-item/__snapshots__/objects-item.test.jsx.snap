// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Objects item component Rendered 1`] = `
<styled.div>
  <styled.svg
    onClick={[Function]}
    viewBox="0 0 32 32"
  >
    <path
      d="M6 32h20l2-22h-24zM20 4v-4h-8v4h-10v6l2-2h24l2 2v-6h-10zM18 4h-4v-2h4v2z"
    />
  </styled.svg>
  . 
</styled.div>
`;
